### PR TITLE
Use callback interface specified in sysl-go instead of generating it each time

### PR DESCRIPTION
**Pull request by gkanz**
Instead of generating the callback interface in `requestrouter.go` (which doesn't change based on generated types), use one specified directly in Go.

Refer to individual commit comments for rationale of the groupings.

Due to the way Go handles interfaces this should not be a breaking change.

### CODE REVIEW

File: codegen/tests/cardspb/cards_api.pb.go [hunk 1]
```diff
@@ -6,12 +6,11 @@ package cardspb
 import (
 	context "context"
 	fmt "fmt"
-	math "math"
-
 	proto "github.com/golang/protobuf/proto"
 	grpc "google.golang.org/grpc"
 	codes "google.golang.org/grpc/codes"
 	status "google.golang.org/grpc/status"
+	math "math"
```
Comment (ChloePlanet) [inline codegen/tests/cardspb/cards_api.pb.go]: This file, `codegen/tests/cardspb/wallet_api.pb.go` and `codegen/tests/simplepb/simplegrpc.pb.go` shouldn't be changed after `goimports -w .` run. Might related to Makefile?

**Review by ChloePlanet — APPROVED** [review]
LGTM

File: codegen/tests/cardspb/cards.pb.go [hunk 2]
```diff
@@ -5,9 +5,8 @@ package cardspb
 
 import (
 	fmt "fmt"
-	math "math"
-
 	proto "github.com/golang/protobuf/proto"
+	math "math"
```
Comment (camscale) [inline codegen/tests/cardspb/cards.pb.go]: i have always run `goimports` on the generated file, which is what gives the original ordering you had. I think it is better to do that. maybe.

**Review by camscale — APPROVED** [review]
LGTM modulo goimports comment.

This is looking much nicer. I like that it is easier to have one place to document the callback interfaces, which we can incrementally improve if any questions come in about it. Removing more lines than being added is another bonus.

:+1:

File: codegen/tests/cardspb/cards_api.pb.go (hunk 1 above)
Comment (gkanz) [reply codegen/tests/cardspb/cards_api.pb.go]: Will fix. Thanks.

File: codegen/tests/cardspb/cards.pb.go (hunk 2 above)
Comment (gkanz) [reply codegen/tests/cardspb/cards.pb.go]: Done.

**Pull request merged**